internal/devbox/devbox.go

```diff
@@ -1007,10 +1007,13 @@ func (d *Devbox) ensureStateIsUpToDateAndComputeEnv(
 
 	// When ensureStateIsUpToDate is called with ensure=true, it always
 	// returns early if the lockfile is up to date. So we don't need to check here
-	if err := d.ensureStateIsUpToDate(ctx, ensure); err != nil && !strings.Contains(err.Error(), "no such host") {
-		return nil, err
+	if err := d.ensureStateIsUpToDate(ctx, ensure); isConnectionError(err) {
+		ux.Fwarning(
+			d.stderr,
+			"Error connecting to the internet. Will attempt to use cached environment.\n",
+		)
```

Comment:
what does the user experience if there was no cached environment? 

i.e. can you `rm -rf .devbox`, turn off internet and see what you experience?

Reply:
It fails with an ugly error. I just added a check so we only keep going if the print dev env cache is there.

There are still a few other edge case failure modes, e.g. if some package has been gc'd from nix store, but I think that's fine.